main.go

```diff
@@ -202,3 +215,37 @@ func wrapAuthHandler(handler http.HandlerFunc, htpasswdFile string, host string)
 	authenticator := auth.NewBasicAuthenticator(host, secrets)
 	return auth.JustCheck(authenticator, handler)
 }
+
+func writePidFile(c *config.Config, accessLogger cache.Logger) error {
+	// create a "pid" directory under cache directory and clean up inactive pid file
+	pidPath := filepath.Join(c.Dir, "pid")
+	err := os.MkdirAll(pidPath, os.FileMode(0744))
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+	err = filepath.Walk(pidPath, func(name string, fileInfo os.FileInfo, err error) error {
+		if !fileInfo.IsDir() {
+			pid, _ := strconv.Atoi(fileInfo.Name())
+			bazelRemoteProcess, err := os.FindProcess(pid)
+			if err != nil {
+				accessLogger.Printf("Error to find pid: %d", bazelRemoteProcess)
+			} else {
+				err = bazelRemoteProcess.Signal(syscall.Signal(0))
+				if err != nil && strings.Contains(err.Error(), "process already finished") {
```

Comment:
Better to check `errno` than pattern match on the error string.

Reply:
That's good idea. But the Signal call only returns error string. Also os.FindProcess(pid) will always returns a pid, so it will always exist

Reply:
`errno` is set after any system call, regardless of what the return value is.

`os.FindProcess()` will always succeed because it's just doing a type-conversion from `int` (the PID) to `*Process`. It makes no guarantees about the existence of the PID.